apps/sms/views/conversation/js/conversation.js

```diff
@@ -2973,52 +2837,51 @@ var ConversationView = {
     return defer.promise;
   },
 
-  discardDraft: function conv_discardDraft() {
-    // If we were tracking a draft properly, update the Drafts object entry.
-    if (!this.draft) {
-      return;
-    }
+  updateDraft() {
+    var conversation = this.activeThread;
 
-    Drafts.delete(this.draft).store();
+    // Need to assimilate recipients in order to check if any entered.
+    this.assimilateRecipients();
 
-    this.draft = null;
-  },
+    // If message content is not empty and/or we have _valid_ recipients in
```

Comment:
We have a bug to save also invalid recipients in https://bugzilla.mozilla.org/show_bug.cgi?id=952519. I also think I have some code in https://bugzilla.mozilla.org/show_bug.cgi?id=1015194 but not sure (that patch will need a big rebase too :/)